client/bench_test.go

```diff
@@ -0,0 +1,210 @@
+func BenchmarkPbInGoNumOut(b *testing.B) {
+	var gn goNumMessage
+
+	for i := 0; i < b.N; i++ {
+		response, err := multiplyOnSiteCPP(initVars.client, &initVars.request)
+		if err != nil {
+			b.Errorf("Couldn't get a response: %v\n", err)
+		}
+		gn.parseResponse(*response)
+	}
+}
+
+func BenchmarkPbInGoPrimitiveOut(b *testing.B) {
+	var pm primitiveMessage
+
+	for i := 0; i < b.N; i++ {
+		response, err := multiplyOnSiteCPP(initVars.client, &initVars.request)
+		if err != nil {
+			b.Errorf("Couldn't get a response: %v\n", err)
+		}
+		pm.parseResponse(*response)
+	}
+}
+
+func BenchmarkGoNumInPbOut(b *testing.B) {
+	var gn goNumMessage
+	gn.parseRequest(initVars.request)
+
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		req := gn.writeRequest()
+		_, err := multiplyOnSiteCPP(initVars.client, &req)
+		if err != nil {
+			b.Errorf("Couldn't get a response: %v\n", err)
+		}
+	}
+}
+
+func BenchmarkGoNumInGoNumOut(b *testing.B) {
+	var gn goNumMessage
+	gn.parseRequest(initVars.request)
+
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		req := gn.writeRequest()
+		response, err := multiplyOnSiteCPP(initVars.client, &req)
+		if err != nil {
+			b.Errorf("Couldn't get a response: %v\n", err)
+		}
+		gn.parseResponse(*response)
+	}
+}
+
+func BenchmarkGoNumInGoPrimitiveOut(b *testing.B) {
+	var pm primitiveMessage
+	var gn goNumMessage
+	gn.parseRequest(initVars.request)
+
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		req := gn.writeRequest()
+		response, err := multiplyOnSiteCPP(initVars.client, &req)
+		if err != nil {
+			b.Errorf("Couldn't get a response: %v\n", err)
+		}
+		pm.parseResponse(*response)
+	}
+}
+
+func BenchmarkGoPrimitiveInPbOut(b *testing.B) {
+	var pm primitiveMessage
+	pm.parseRequest(initVars.request)
+
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		req := pm.writeRequest()
+		_, err := multiplyOnSiteCPP(initVars.client, &req)
+		if err != nil {
+			b.Errorf("Couldn't get a response: %v\n", err)
+		}
+	}
+}
+
+func BenchmarkGoPrimitiveInGoNumOut(b *testing.B) {
+
+	var gn goNumMessage
+	var pm primitiveMessage
+	pm.parseRequest(initVars.request)
+
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		req := pm.writeRequest()
+		response, err := multiplyOnSiteCPP(initVars.client, &req)
+		if err != nil {
+			b.Errorf("Couldn't get a response: %v\n", err)
+		}
+		gn.parseResponse(*response)
+	}
+}
+
+func BenchmarkGoPrimitiveInGoPrimitiveOut(b *testing.B) {
+	var pm primitiveMessage
+	pm.parseRequest(initVars.request)
+
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		req := pm.writeRequest()
+		response, err := multiplyOnSiteCPP(initVars.client, &req)
+		if err != nil {
+			b.Errorf("Couldn't get a response: %v\n", err)
+		}
+		pm.parseResponse(*response)
+	}
+}
+
+func BenchmarkMultiplyOnSiteGO(b *testing.B) {
+	var gn goNumMessage
+	gn.parseRequest(initVars.request)
+
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		gn.Product.Mul(gn.Multiplier, gn.Multiplicand)
+	}
+}
+
+func TestParseRequestAndWriteRequest(t *testing.T) {
```

Comment:
It would be nice to have this check after each benchmark loop. Smoke testing as such.